analysis_driver/report_generation/report_crawlers.py

```diff
@@ -74,38 +74,33 @@ def _update_doc_set(d, k, v):
         d[k].add(v)
 
     def _populate_barcode_info_from_sample_sheet(self, samplesheet):
+        """
+        :param analysis_driver.reader.SampleSheet samplesheet:
+        """
         self.barcodes_info = defaultdict(dict)
         self.unexpected_barcodes = {}
         self.libraries = defaultdict(dict)
         self.lanes = defaultdict(dict)
         self.run = {ELEMENT_RUN_NAME: self.run_id, ELEMENT_RUN_ELEMENTS: []}
         self.projects = defaultdict(dict)
 
-        for project_id, proj_obj in samplesheet.sample_projects.items():
+        for project_id, proj_obj in samplesheet.projects.items():
```

Comment:
We should use `samplesheet.projects.values()` here and get the project id from `proj_obj.name`